digital-twin-model/src/sdv_v2.rs

```diff
@@ -0,0 +1,621 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+// This file contains the generated code for the Software Defined Vehicle (SDV) model.
+// This code is manually generated today, but in the future it should be automatically generated from the DTDL.
+
+/// The context value for all JSON-LD generated by the code in this file.
+fn context() -> Vec<String> {
+    vec!["dtmi:dtdl:context;2".to_string(), "dtmi:sdv:context;2".to_string()]
+}
+
+/// Constants and type defintiions (including JSON serialization/deserialization) for each interface in the SDV model.
+#[allow(dead_code)]
+pub mod airbag_seat_massager {
+    pub const ID: &str = "dtmi:sdv:airbag_seat_massager;1";
+    pub fn id() -> String {
+        ID.to_string()
+    }
+    pub const DESCRIPTION: &str = "Airbag Seat Massager Interface.";
+    pub mod store_sequence {
+        pub const ID: &str = "dtmi:sdv:airbag_seat_massager:store_sequence;1";
+        pub fn id() -> String {
+            ID.to_string()
+        }
+        pub const NAME: &str = "store_sequence";
+        pub const DESCRIPTION: &str = "Store a massage sequence.";
+        pub mod request {
+            pub const ID: &str = "dtmi:sdv:airbag_seat_massager:store_sequence:request;1";
+            pub fn id() -> String {
+                ID.to_string()
+            }
+            pub const NAME: &str = "request";
+            pub const DESCRIPTION: &str = "Request.";
+            #[derive(derivative::Derivative)]
+            #[derivative(Default)]
+            #[derive(serde_derive::Serialize, serde_derive::Deserialize, Debug)]
+            pub struct TYPE {
```

Comment:
Because there are multiple pieces of meta data for each of these entities, e.g. name, mode id, description, and type.

Reply:
This is not incompatible with renaming the struct. We can call it something more descriptive and retain these fields

Reply:
The thing that makes this difficult to use is that there are a number of types with the same name `TYPE`. If we gave them all more descriptive names such as `StoreSequenceRequest`, `StoreSequenceResponse`, `PerformStepRequest`, `PerformStepResponse`, and so on it will make them much easier to use. As-is this requires that you qualify or alias the types to use them, and if I were implementing something with this contract I would almost certainly alias these types to help with readability

Reply:
In Rust, you can't mix constant definitions with struct definitions.

I can define the structs outside and use TYPE to alias its respective struct (to retain a connection with the respective NAME, ID and DESCRIPTION metadata.  How does sound?

Reply:
I'm not sure I understand what you're saying. To me it looks like you've mixed constant definitions and struct definitions here. The `airbag_seat_massager::store_sequence::request` module contains definitions for both consts (`ID`, `NAME`, `DESCRIPTION`) and structs (`TYPE`) mixed together.

You've mentioned metadata a few times now, and I know that this file is supposed to mirror the content of DTDL files. Is the intent that this is called `TYPE` to mirror the DTDL `@type` definition? I'm not a fan of this convention, and would suggest that we translate it into something that feels more comfortable in rust for the reasons I mentioned above.

That said, I think your suggestion (define the types elsewhere and alias them here) would satisfy this and allow you to keep the more direct association with the DTDL if that's desirable.

Reply:
Sorry, I meant in Rust that struct definitions cannot contain const definitions.

The idea with this metadata (constants and types) is to organize it in a manner that makes it easy for a developer to find the metadata that they need.  What is important here is the namespace organization rather than the file organization.  Ideally, all of the metadata for each DTDL entity should be found in the same namespace.

Reply:
WIlliam and I chatted about this.  The command requests and responses may later have their types defined in the schemas, which would allow the TYPE constant to alias these rather than define the type/struct inline.  This does not need to be fixed now but should be considered as a possible improvement.